include/deal.II/base/bounding_box.h

```diff
@@ -77,14 +102,23 @@ class BoundingBox
   const std::pair<Point<spacedim,Number>,Point<spacedim,Number>> &get_boundary_points () const;
 
   /**
+   * Check if the current object and @p other_bbox are neighbors, i.e. if the boxes
+   * have dimension spacedim, check if their insersection is non empty
+   *
+   * Return an enumerator of type NeighborType.
+   */
+  NeighborType is_neighbor(const BoundingBox<spacedim,Number> &other_bbox) const;
```

Comment:
I though a bit about this... there are reasons to change the name, but mainly the idea behind the function is also: 0 means "no intersection", non-zero means intersection (and then the idea is that "intersection" can be classified in different ways).

Otherwise I can propose the name neighbor_kind for instance?